### DISCUSSION TITLE
Custom std::wstring serialization/deserialization

### DISCUSSION THREAD

**Discussion by paskulinb** [nlohmann/json#2702] [Q&A] [answered]
I would prefer the library to convert every std::wstring to utf8 encoded std::string (narrow string).
What would be the best way (if any) to make this conversion (both ways; set and get) automatic by the library?
I will provide the conversion functions ucs<->utf8.

Best regards,
Borut

**Comment by nlohmann** [comment, marked as answer]
I'd rather have this quite generic conversion between two C++ types outside of this library. JSON uses UTF-8 as default encoding, so using `std::string` is an easy choice. If, however, this conversion is simple and painless, then only the appropriate `to_json` and `from_json` methods need to be provided.

**Reply by paskulinb** [reply]
Thank you for the reply.
If I understand your hint, the `to_json` and `from_json` will override the default behavior for `std::wstring`. Is it correct?

Yet another question/idea:
If I use `typedef std::wstring my_string`, will the library consider it a different type, or does it use the underlaying type?

I use `std::wstring` everywhere in my application, but when going 'outside' (file, network), I convert everything to UTF-8 encoded `std::string`. That is why I am asking this.
